sqlmesh/core/config/root.py

```diff
@@ -287,6 +287,11 @@ def default_gateway_name(self) -> str:
 
     @property
     def dialect(self) -> t.Optional[str]:
+        if self.default_gateway:
+            gateway_config = self.gateways[self.default_gateway]
+            if gateway_config.model_defaults:
+                return gateway_config.model_defaults.dialect
+
```

Comment:
I'm not sure if this is the right way, why just do it for the dialect? Folks may want to choose different defaults per gateway in general.

In an ideal world, imo `Config.model_defaults` should be the result of merging the base / global defaults with the gateway's defaults, prioritizing the latter when there's a key collision.

@tobymao thoughts?

Reply:
right i agree